sympy/solvers/solveset.py

```diff
@@ -1201,24 +1200,21 @@ def _transolve(f, symbol, domain):
     How to add new class of equations
     =================================
 
-    `\_transolve` is designed in such a way that it becomes an easy
-    task to add a new class of equation solver.
-
-    The first task that needs to be done for adding your own solver
-    is to decide from where its *identification* *helper* will be
-    invoked. To do so determine the general form of the class of the
-    equation, for example the general form of the exponential equation
-    is `a*f(x) + b*g(x)` where `f(x)` and `g(x)` are power terms, so we
-    need to place the invocation condition inside `Add` case. Once the
-    place for *identification* *helper* is determined, you can add a
-    call to its *solving* *helper*.
-
-    For your class of equation you need to define your own
-    *identification* and *solving* *helpers*. The *identification*
-    *helper* should return either `True` if the given equation belongs
-    to the class otherwise `False`. *Solving* *helpers* should return
-    either the exact solution or a tractable form which `solveset`
-    can handle.
+    The design of `\_transolve` makes adding a new class of equation
+    solver easy.
+
+    Decide from where the *identification* *helper* will be invoked.
```

Comment:
Is there any transcendental equation (other than something like cos(x) = 0) that can be solved/handled by transolve that is not an Add? 

This paragraph is confusing with the example put in the middle of the instructions. "Once done" and "add a call" are not clearly defined. It would be better to give a code snippet like

```python
if eq.is_Add:
    ...
    if _is_expo(eq, x):
        new_eq = _solve_expo(eq, x)
    ...
```

Also, what to name the result should probably be specified. Are different names used for the case when a solution is returned vs the case where a rewritten form is returned?

Reply:
> Is there any transcendental equation (other than something like cos(x) = 0) that can be solved/handled by transolve that is not an Add?

I can't think of any other than `Add` as of now if there are we can create another internal function for it.

>It would be better to give a code snippet like

I will add the code snippet for better understanding.

> Are different names used for the case when a solution is returned vs the case where a rewritten form is returned?

Yes, when the equation is recasted then `new_eq` will be updated and then `solveset` will be called to update the `result`, whereas when solution is returned the `result` is directly updated